FirebaseAI/Sources/GenerativeAIService.swift

```diff
@@ -207,6 +210,30 @@ struct GenerativeAIService {
     return urlRequest
   }
 
+  private func fetchAppCheckToken(appCheck: AppCheckInterop) async throws
+    -> FIRAppCheckTokenResultInterop {
+    if aiConfig.appCheck.requireLimitedUseTokens {
+      if let token = await appCheck.getLimitedUseToken?() {
+        return token
+      }
+
+      let errorMessage =
+        "The provided App Check token provider doesn't implement getLimitedUseToken(), but requireLimitedUseTokens was enabled."
+
+      #if Debug
+        fatalError(errorMessage)
+      #else
+        throw NSError(
+          domain: "\(Constants.baseErrorDomain).\(Self.self)",
+          code: AILog.MessageCode.appCheckTokenFetchFailed.rawValue,
+          userInfo: [NSLocalizedDescriptionKey: errorMessage]
+        )
+      #endif
+    }
```

Comment:
Do you know how this error gets wrapped by Firebase AI Logic? NSError is mainly for an Objective-C context. Is it easy to get the error message out if your at the top-level trying to parse the internal error?

cc: @andrewheard

Reply:
This gets bubbled up into internal error. If you print it, you see the domain, code, and user info. 

If you catch the `GenerativeModelError.internal(let underlying)` case, you can underlying.localizedDescription to get the error message